packages/lisk-p2p/src/response_handler.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import { isAlpha, isIP, isNumeric, isPort } from 'validator';
+import { InvalidPeerAddress } from './errors';
+import { PeerConfig } from './peer';
+
+const IPV4_NUMBER = 4;
+const IPV6_NUMBER = 6;
+
+interface RawPeerObject {
+	readonly height?: string;
+	readonly ip: string;
+	readonly os?: string;
+	readonly version?: string;
+	readonly wsPort: string;
+}
+
+export const checkPeerAddress = (peer: unknown): boolean => {
+	if (!peer) {
+		return false;
+	}
+
+	const { ip, wsPort } = peer as RawPeerObject;
+
+	if (!ip || !wsPort) {
+		return false;
+	}
+
+	if ((!isIP(ip, IPV4_NUMBER) && !isIP(ip, IPV6_NUMBER)) || !isPort(wsPort)) {
+		return false;
+	}
+
+	return true;
+};
+
+export const instantiatePeerFromResponse = (
+	peer: unknown,
+): PeerConfig | Error => {
+	if (!checkPeerAddress(peer)) {
+		return new InvalidPeerAddress(`Invalid Peer Ip or Port`);
+	}
+
+	const rawPeer = peer as RawPeerObject;
+
+	const os = rawPeer.os && isAlpha(rawPeer.os.toString()) ? rawPeer.os : '';
+	const version =
+		rawPeer.version && isAlpha(rawPeer.version.toString())
+			? rawPeer.version
```

Comment:
maybe we should throw if version is empty. (maybe this relates to hand shake).
Please confirm with the core

Reply:
I will check this with Jon and core team

Reply:
as discussed, I will use `semver` that we use in core to validate version and throw the error if invalid